tests/integration/test_workflow_yaml_validation.py

```
@@ -0,0 +1,500 @@
+"""
+Advanced YAML validation tests for GitHub workflow files.
+
+This module provides comprehensive validation of YAML structure,
+syntax, and best practices for all workflow files.
+"""
+
+import pytest
+import yaml
+from pathlib import Path
+from typing import Dict, Any, List
```

Ruff (0.14.6)

`typing.Dict` is deprecated, use `dict` instead

(UP035)